sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/BlobEncryptionPolicy.java

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.cryptography;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.models.Metadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.ShortBufferException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a blob encryption policy that is used to perform envelope encryption of Azure storage blobs.
+ * <a href="https://docs.microsoft.com/en-us/azure/storage/common/storage-client-side-encryption-java?toc=%2fazure%2fstorage%2fblobs%2ftoc.json">Azure Docs</a>
+ * for more information.
+ */
+final class BlobEncryptionPolicy {
```

Comment:
I wonder if we should rename this now that "policy" has become overloaded. Perhaps just BlobEncryptor? Or even just moving the methods on here to EncryptedBlockBlobClient might work. Since this is an internal type, and the methods are only used within EncryptedBlockBlobClient, and this is never really passed around, perhaps a separate type is not even necessary. Probably the reason it's been contained in this type for v11 and track 2 is just because that's what we did in track 1.

Reply:
Moving to encryptedblockblobasyncclient